pandas/_libs/algos.pyx

```diff
@@ -1099,43 +1101,35 @@ def rank_2d(
     argsorted = _as.astype('i8')
 
     for i in range(n):
-        if rank_t is object:
-            dups = sum_ranks = infs = 0
-        else:
-            dups = sum_ranks = 0
+        dups = sum_ranks = infs = 0
 
         total_tie_count = 0
         count = 0.0
         for j in range(k):
-            if rank_t is not object:
-                sum_ranks += j + 1
-                dups += 1
-
             val = values[i, j]
 
-            if rank_t is not uint64_t:
-                if rank_t is object:
-                    skip_condition = (val is nan_value) and keep_na
-                else:
-                    skip_condition = (val == nan_value) and keep_na
-                if skip_condition:
-                    ranks[i, argsorted[i, j]] = NaN
-
-                    if rank_t is object:
-                        infs += 1
-
-                    continue
+            if mask[i, argsorted[i, j]] and keep_na:
+                ranks[i, argsorted[i, j]] = NaN
```

Comment:
could define `idx = argsoted[i, j]` on L1110 to avoid doing the lookup twice

Reply:
Thanks